### ISSUE TITLE
mq_pcf.py RFH2 issue

### ISSUE THREAD

**Issue by rkbroder** [colinpaicemq/MQTools#1]
Ran the MQ_PCF.PY code against a message from the new v9.x System Monitoring published stats. Had an issue with RFH2. Saw in the code you had commented out processing of that. Then got it to run but it does not seem to like the data. It maybe my version of Python (ie v20)

In v9.1 I created a subscription as:

$SYS/MQ/INFO/QMGR/BOBBEE/Monitor/DISK/Log

and dropped it to a queue and tried to read it with mq_pcf.py

C:\Python38\python.exe D:/@Document/PYTHON/MQ_DEMO/PCF/mq_pcf.py -qm BOBBEE -channel SYSTEM.ADMIN.SVRCONN -conname localhost(1714) -userid rkbroder -password K@riann8 -queue BOBBEE_OUT
============================== <class 'str'>
{'StrucId': 'MD', 'Version': 1, 'Report': 'NONE', 'MsgType': 'DATAGRAM', 'Expiry': 33452, 'Feedback': 'NONE', 'Encoding': 546, 'CodedCharSetId': 437, 'Format': 'MQHRF2', 'Priority': 0, 'Persistence': 'NOT_PERSISTENT', 'MsgId': '0x414d5120424f424245452020202020204ca5d75e00a8a320', 'CorrelId': '0x414d5120424f424245452020202020204ca5d75e20a3a790', 'BackoutCount': 0, 'ReplyToQ': '', 'ReplyToQMgr': 'BOBBEE', 'UserIdentifier': 'BOBBEEBRODER', 'AccountingToken': '0x0000000000000000000000000000000000000000000000000000000000000000', 'ApplIdentityData': '', 'PutApplType': 'QMGR_PUBLISH', 'PutApplName': 'BOBBEE', 'PutDate': '20200603', 'PutTime': '18270985', 'ApplOriginData': '', 'GroupId': '0x000000000000000000000000000000000000000000000000', 'MsgSeqNumber': 1, 'Offset': 0, 'MsgFlags': 0, 'OriginalLength': -1}
problem with  b'\x15\x00\x00\x00$\x00\x00\x00\x03\x00\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x0c\x00\x00\x00\x04\x00\x00\x00D\x00\x00\x00\xdf\x07\x00\x00R\x03\x00\x000\x00\x00\x00BOBBEE                                          \x03\x00\x00\x00\x10\x00\x00\x00G\x03\x00\x00\x01\x00\x00\x00\x03\x00\x00\x00\x10\x00\x00\x00H\x03\x00\x00\x02\x00\x00\x00\x17\x00\x00\x00\x18\x00\x00\x00M\x03\x00\x00\x01\x00\x00\x00\xb6I \x08\x00\x00\x00\x00\x17\x00\x00\x00\x18\x00\x00\x00\x00\x00\x00\x00\xd2U\x02\x00\x00\xc0\xf4\t\x00\x00\x00\x00\x17\x00\x00\x00\x18\x00\x00\x00\x01\x00\x00\x00\x10\x00\x00\x00\x00\xc0\xeb\x11\x00\x00\x00\x00\x17\x00\x00\x00\x18\x00\x00\x00\x04\x00\x00\x00\xd2U\x02\x00\x00\x00\x00\x00\x00\x00\x00\x00\x17\x00\x00\x00\x18\x00\x00\x00\x05\x00\x00\x00\x03\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x03\x00\x00\x00\x10\x00\x00\x00\x06\x00\x00\x00\xf0\x04\x00\x00\x03\x00\x00\x00\x10\x00\x00\x00\x0c\x00\x00\x00\xe4\x14\x00\x00\x03\x00\x00\x00\x10\x00\x00\x00\x07\x00\x00\x00p\x00\x00\x00\x03\x00\x00\x00\x10\x00\x00\x00\x08\x00\x00\x00p\x00\x00\x00'
{"sReason": "????", "MQMD": {"StrucId": "MD", "Version": 1, "Report": "NONE", "MsgType": "DATAGRAM", "Expiry": 33452, "Feedback": "NONE", "Encoding": 546, "CodedCharSetId": 437, "Format": "MQHRF2", "Priority": 0, "Persistence": "NOT_PERSISTENT", "MsgId": "0x414d5120424f424245452020202020204ca5d75e00a8a320", "CorrelId": "0x414d5120424f424245452020202020204ca5d75e20a3a790", "BackoutCount": 0, "ReplyToQ": "", "ReplyToQMgr": "BOBBEE", "UserIdentifier": "BOBBEEBRODER", "AccountingToken": "0x0000000000000000000000000000000000000000000000000000000000000000", "ApplIdentityData": "", "PutApplType": "QMGR_PUBLISH", "PutApplName": "BOBBEE", "PutDate": "20200603", "PutTime": "18270985", "ApplOriginData": "", "GroupId": "0x000000000000000000000000000000000000000000000000", "MsgSeqNumber": 1, "Offset": 0, "MsgFlags": 0, "OriginalLength": -1}, "msg": {"body": "0x15000000240000000300000000000000010000000100000000000000000000000c0000000400000044000000df0700005203000030000000424f42424545202020202020202020202020202020202020202020202020202020202020202020202020202020202020030000001000000047030000010000000300000010000000480300000200000017000000180000004d03000001000000b649200800000000170000001800000000000000d255020000c0f409000000001700000018000000010000001000000000c0eb1100000000170000001800000004000000d25502000000000000000000170000001800000005000000030000000000000000000000030000001000000006000000f004000003000000100000000c000000e41400000300000010000000070000007000000003000000100000000800000070000000"}, "time": "2020-06-03 13:31:24.668537", "delta": "0:00:00"}
============================== <class 'str'>
{'StrucId': 'MD', 'Version': 1, 'Report': 'NONE', 'MsgType': 'DATAGRAM', 'Expiry': 33552, 'Feedback': 'NONE', 'Encoding': 546, 'CodedCharSetId': 437, 'Format': 'MQHRF2', 'Priority': 0, 'Persistence': 'NOT_PERSISTENT', 'MsgId': '0x414d5120424f424245452020202020204ca5d75e00aba320', 'CorrelId': '0x414d5120424f424245452020202020204ca5d75e20a3a790', 'BackoutCount': 0, 'ReplyToQ': '', 'ReplyToQMgr': 'BOBBEE', 'UserIdentifier': 'BOBBEEBRODER', 'AccountingToken': '0x0000000000000000000000000000000000000000000000000000000000000000', 'ApplIdentityData': '', 'PutApplType': 'QMGR_PUBLISH', 'PutApplName': 'BOBBEE', 'PutDate': '20200603', 'PutTime': '18271985', 'ApplOriginData': '', 'GroupId': '0x000000000000000000000000000000000000000000000000', 'MsgSeqNumber': 1, 'Offset': 0, 'MsgFlags': 0, 'OriginalLength': -1}
problem with  b'\x15\x00\x00\x00$\x00\x00\x00\x03\x00\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x0c\x00\x00\x00\x04\x00\x00\x00D\x00\x00\x00\xdf\x07\x00\x00R\x03\x00\x000\x00\x00\x00BOBBEE                                          \x03\x00\x00\x00\x10\x00\x00\x00G\x03\x00\x00\x01\x00\x00\x00\x03\x00\x00\x00\x10\x00\x00\x00H\x03\x00\x00\x02\x00\x00\x00\x17\x00\x00\x00\x18\x00\x00\x00M\x03\x00\x00\x01\x00\x00\x00\\\xb6\x98\x00\x00\x00\x00\x00\x17\x00\x00\x00\x18\x00\x00\x00\x00\x00\x00\x00\xd2U\x02\x00\x00\xc0\xf4\t\x00\x00\x00\x00\x17\x00\x00\x00\x18\x00\x00\x00\x01\x00\x00\x00\x10\x00\x00\x00\x00\xc0\xeb\x11\x00\x00\x00\x00\x17\x00\x00\x00\x18\x00\x00\x00\x04\x00\x00\x00\xd2U\x02\x00\x00\x10\x00\x00\x00\x00\x00\x00\x17\x00\x00\x00\x18\x00\x00\x00\x05\x00\x00\x00\x03\x00\x00\x00t\x07\x00\x00\x00\x00\x00\x00\x03\x00\x00\x00\x10\x00\x00\x00\x06\x00\x00\x00\x97\x05\x00\x00\x03\x00\x00\x00\x10\x00\x00\x00\x0c\x00\x00\x00\xc4\x14\x00\x00\x03\x00\x00\x00\x10\x00\x00\x00\x07\x00\x00\x00p\x00\x00\x00\x03\x00\x00\x00\x10\x00\x00\x00\x08\x00\x00\x00p\x00\x00\x00'
{"sReason": "????", "MQMD": {"StrucId": "MD", "Version": 1, "Report": "NONE", "MsgType": "DATAGRAM", "Expiry": 33552, "Feedback": "NONE", "Encoding": 546, "CodedCharSetId": 437, "Format": "MQHRF2", "Priority": 0, "Persistence": "NOT_PERSISTENT", "MsgId": "0x414d5120424f424245452020202020204ca5d75e00aba320", "CorrelId": "0x414d5120424f424245452020202020204ca5d75e20a3a790", "BackoutCount": 0, "ReplyToQ": "", "ReplyToQMgr": "BOBBEE", "UserIdentifier": "BOBBEEBRODER", "AccountingToken": "0x0000000000000000000000000000000000000000000000000000000000000000", "ApplIdentityData": "", "PutApplType": "QMGR_PUBLISH", "PutApplName": "BOBBEE", "PutDate": "20200603", "PutTime": "18271985", "ApplOriginData": "", "GroupId": "0x000000000000000000000000000000000000000000000000", "MsgSeqNumber": 1, "Offset": 0, "MsgFlags": 0, "OriginalLength": -1}, "msg": {"body": "0x15000000240000000300000000000000010000000100000000000000000000000c0000000400000044000000df0700005203000030000000424f42424545202020202020202020202020202020202020202020202020202020202020202020202020202020202020030000001000000047030000010000000300000010000000480300000200000017000000180000004d030000010000005cb6980000000000170000001800000000000000d255020000c0f409000000001700000018000000010000001000000000c0eb1100000000170000001800000004000000d255020000100000000000001700000018000000050000000300000074070000000000000300000010000000060000009705000003000000100000000c000000c41400000300000010000000070000007000000003000000100000000800000070000000"}, "time": "2020-06-03 13:31:24.668537", "delta": "0:00:00"}
message not found

Process finished with exit code 0

**Comment by colinpaicemq** [comment]
HI,

I'll look into it.  You might try altering the queue to have PROPCTL(NONE)
and see if that makes a difference
regards

Colin

**Comment by rkbroder** [comment]
I updated the issue. Think it did not take or got there late.

You had a line of commented code that referenced RFH2. I fixed and ran 
again. It digested the RFH2. But is stuck on the BYTE array from the data 
portion. I tried converting to STRING which I found is the usual approach. 
It choked on it around byte 44. I posted to MQSERIES.net. Morag was able 
to convert. But she neglected to say what she did.
 
                                Bobbee

Failures, repeated failures, are finger posts on the road to achievement. 
One fails forward toward success. 

9174536790, rkbroder@us.ibm.com
